migrate/indexerdb0to1.js

```diff
@@ -0,0 +1,206 @@
+  console.log('Migrating filters..');
+  let parent  = db.batch();
+
+  const iter = db.iterator({
+    gte: layout.h.min(),
+    lte: layout.h.max(),
+    keys: true,
+    values: true
+  });
+
+  const migratedDB = bdb.create({
+    location: path.join(db.location, 'BASIC'),
+    memory: false,
+    compression: true,
+    createIfMissing: true
+  });
+
+  await migratedDB.open();
+
+  let parentMigratedDB = migratedDB.batch();
+
+  let migratedFilters = 0;
+
+  await iter.each(async (height, hash) => {
+    const rawFilter = await blockStore.readFilter(hash, filters.BASIC);
+    const filter = Filter.fromRaw(rawFilter);
+    const filterHash = hash256.digest(filter.filter);
+
+    parentMigratedDB.put(height, hash);
+    parent.del(height);
+
+    parentMigratedDB.put(layout.f.encode(hash), filterHash);
+    parent.del(layout.f.encode(hash));
+
+    migratedFilters += 1;
+
+    if (migratedFilters % 10000 === 0) {
+      console.log('migrated %d filters.', migratedFilters);
+      await parentMigratedDB.write();
+      await parent.write();
+
+      parentMigratedDB = migratedDB.batch();
+      parent = db.batch();
+    }
+  });
+
+  let raw = await db.get(layout.R.encode());
+  parentMigratedDB.put(layout.R.encode(), raw);
+  parent.del(layout.R.encode());
+
+  raw = await db.get(layout.O.encode());
+  parentMigratedDB.put(layout.O.encode(), raw);
+  parent.del(layout.O.encode());
+
+  const version = await db.get(layout.V.encode());
+  parentMigratedDB.put(layout.V.encode(), version);
+  parent.del(layout.V.encode());
+
+  await parentMigratedDB.write();
+  await parent.write();
+
+  console.log('%d filters migrated.', migratedFilters);
+
+  await db.close();
+  await db.destroy();
+
+  await migratedDB.close();
+}
+
+/*
+ * Execute
+ */
+
+let count = 0;
+
+(async () => {
+  const indexesLocation = path.resolve(process.argv[2]);
+  const indexes = ['tx', 'addr', 'filter'];
+
+  const compact = true;
+
+  for (const index of indexes) {
+    const location = path.join(indexesLocation, index);
+    if (!(await fs.exists(location))) {
+      continue;
+    }
+
+    const db = bdb.create({
+      location: location,
+      memory: false,
+      compression: true,
+      createIfMissing: false
+    });
+
+    console.log('Opened %s.', location);
+    await db.open();
+
+    const version = await getVersion(db);
+
+    switch (version) {
+      case 0:
+        if (index === 'filter') {
+          await blockStore.ensure();
+          await blockStore.open();
+          await migrateFilter(db);
+          const filterDB = bdb.create({
+            location: path.join(location, 'BASIC'),
+            memory: false,
+            compression: true,
+            createIfMissing: false
+          });
+          await filterDB.open();
+          await updateVersion(filterDB, 1);
+          count += 1;
+
+          if (compact) {
+            console.log('Compacting database.');
+            await filterDB.compactRange();
+          }
```

Comment:
Let's remove all the compaction for now. Migrating a fully synced mainnet indexer set WITH compaction took my machine > 6 hours. I ran the script a second time with these lines pulled out and complete migration finished in about 5 minutes!